db/warehouse/migrate/20230525202043_remove_name_uniqueness_index.rb

```diff
@@ -0,0 +1,9 @@
+class RemoveNameUniquenessIndex < ActiveRecord::Migration[6.1]
+  def up
+    remove_index :CustomClientName, name: "unique_index_ensuring_one_primary_per_client"
```

Comment:
Apparently we just cannot have this, because rails can't handle validation if we update attributes in a way that destroys the primary and adds a new primary.
Kind of like https://stackoverflow.com/questions/2772236/validates-uniqueness-of-in-destroyed-nested-model-rails

I moved the validation to happen in the Client. If client has >1 primary name, its a problem.